### ISSUE TITLE
Rename `Comment.comment`

### ISSUE THREAD

**Issue by oliverklee** [sabberworm/PHP-CSS-Parser#450]
Maybe `Comment.contents`  or `Comment.text`?

**Comment by JakeQZ** [comment]
> Maybe `Comment.contents` or `Comment.text`?

`DOMDocument` uses `textContent`, which suggests using `sContent` here to be consistent with both that and the Hungarian notation used in this project.

**Comment by oliverklee** [comment]
I'd like to get rid of the Hungarian notation sooner or later (now that we can use native type declarations and PHPDoc annotations). However, how we name the property isn't that important as long as we encapsulate it. So should we name the getter/setter `getTextContent` and `setTextContent`?

**Comment by sabberworm** [comment]
`textContent` in DOM is a very generic property, applicable to all sorts of node types, not just comments. I would be fine with either `text` or `contents` but I also don’t really see a problem with keeping the `comment` TBH.

**Comment by JakeQZ** [comment]
My concern with renaming properties and methods, particularly if they are `public` (though also if they are `protected`), is that it would be a breaking change.

PHPUnit renamed some of the assertion methods around version 8 or 9.  (IIRC the old names were deprecated in version 8, and removed in version 10.)  This prevented us using the latest version when testing with the latest PHP, while we still needed older versions compatible with older PHP versions we still supported (without special-casing the version of PHPUnit and calling the appropriate method for the version, which we did not do).  WordPress has occasionally done similar renames, though does generate PHP warnings when deprecated functions and methods are called.

In short: such renames generate extra work for users wishing to upgrade, and can also result in dependency and version-compatibility problems that are difficult to resolve.

We'd first have to mark the old method/property as deprecated, with its implementation chaining on to the newly named equivalent (for properties, the magic `__get` and `__set` can be used).  Then in some later release (perhaps 3 years later), finally remove the deprecated one.

The question I'd ask: Does the overall cost (including that to users) perhaps outweigh the benefit?

**Comment by oliverklee** [comment]
This is the general deprecation policy I'd like to suggest both for code removal and for renaming public things:

1. create the new method(s) (if there is a replacement) and redirect the old methods to the new ones
2. create a ticket in the next breaking milestone x.0.0 after the next release: "Drop it" which gets the ID yyy
3. add `@deprecated yyy will be removed in x.0.0" annotations to the old method(s)
4. once we're working on x.0.0, drop the old method(s)

This allows users of our package to keep using the old method(s) for some time and automatically scan their code for usages of the deprecated methods (using their IDE or a [PHPStan ruleset](https://github.com/phpstan/phpstan-deprecation-rules)).

Would this work for you?

And yes, I think having good, clear, easy-to-understand and hard-to-misunderstand names is immensely important and often makes this worth the hassle.
